app/screens/thread/thread_follow_button/thread_follow_button.tsx

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {Platform, StyleSheet, TouchableOpacity, View} from 'react-native';
+
+import {updateThreadFollowing} from '@actions/remote/thread';
+import FormattedText from '@components/formatted_text';
+import {useServerUrl} from '@context/server';
+import {useTheme} from '@context/theme';
+import {preventDoubleTap} from '@utils/tap';
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+
+import type ThreadModel from '@typings/database/models/servers/thread';
+
+type Props = {
+    teamId: string;
+    thread: ThreadModel;
+};
+
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => {
+    return {
+        container: {
+            borderColor: theme.sidebarHeaderTextColor,
+            borderWidth: StyleSheet.hairlineWidth,
+            borderRadius: 4,
+            paddingVertical: 4.5,
+            paddingHorizontal: 10,
+            opacity: 0.72,
+            ...Platform.select({
+                android: {
+                    marginRight: 12,
+                },
+                ios: {
+                    right: -4,
+                },
+            }),
+        },
+        containerActive: {
+            backgroundColor: changeOpacity(theme.sidebarHeaderTextColor, 0.24),
+            borderColor: 'transparent',
+            opacity: 1,
+        },
+        text: {
+            color: theme.sidebarHeaderTextColor,
+            fontWeight: '600',
+            fontSize: 12,
+        },
+    };
+});
+
+function ThreadFollow({teamId, thread}: Props) {
+    const theme = useTheme();
+    const styles = getStyleSheet(theme);
+
+    const serverUrl = useServerUrl();
+
+    const onPress = preventDoubleTap(() => {
+        updateThreadFollowing(serverUrl, teamId, thread.id, !thread.isFollowing);
+    });
+
+    const containerStyle = [styles.container];
```

Comment:
I earlier had it and later removed it based on the feedback of having it could be overkill as this is intended to change only when `following` state is changed. I'm making a change to the observable to get only the `following state` and not the whole thread.